### ISSUE TITLE
[Epic] Fix frontend with Kubeflow 1.3

### ISSUE THREAD

**Issue by wg102** [StatCan/jupyter-apis#68]
After a few decisions, it was decided that CWA was postponed for now, in favor of speed, jupyter-apis will be upgrade with 1.3 kubeflow frontend.  See https://github.com/StatCan/daaas/issues/868 and https://github.com/StatCan/daaas/issues/831 for details. 

As of now, the backend was done. [here] (https://github.com/StatCan/jupyter-apis/commit/863a996748b37e94aae013ea9de3d640af530423)
What is left is to add our own customization to the front end. 

The following is the list of commits relating to the following customizations, and list them under the related feature, originaly [here](https://github.com/StatCan/daaas/issues/831#issuecomment-1041597135). An issue will be created for each checkbox.

## Front end customizations:
- [x] #76  - Jose + Rohan
  - [Merge pull request #19 from StatCan/notebook-server-validation](https://github.com/StatCan/jupyter-apis/commit/22bdc135476e5eaa17318bf635d79aa5890ad888)
    - [fix: notebook server validation](https://github.com/StatCan/jupyter-apis/pull/19/commits/1e666335ee1f08251652335eacb5e721d4fd0870) -> the form-name.component.html seems to have changed from what we had, to whats in 1.3 significantly.

    - [fix: Fix cross-validation issues, error messages](https://github.com/StatCan/jupyter-apis/pull/19/commits/32eb310e955fd2e02ea8cec18edfd56ceea11827)
    - [fix: Gi for RAM not optional](https://github.com/StatCan/jupyter-apis/pull/19/commits/9cbb66cb47c31ff516c8ea89f11551022b2aa4bb) 
    
- [x]  #80 -Rohan
  - [Merge pull request #12 from StatCan/fix-kubeflow-collision-information](https://github.com/StatCan/jupyter-apis/commit/5caa0a449bf31913e88f241c343cb5578810ba87)
    - [fix: kubeflow collision](https://github.com/StatCan/jupyter-apis/commit/f068e8d34f8ad4f0768a0baa078cb838d0711130)
  - [Merge pull request #38 from StatCan/fix-max-limits](https://github.com/StatCan/jupyter-apis/commit/344d4fcae658d5136bbc772896025d589ad43873)
    - [fix: update ram limits](https://github.com/StatCan/jupyter-apis/pull/38/commits/f8c29eecb38e96b24933fa7932497218cb47ef60)
  - [Merge pull request #45 from StatCan/fix-gpu-specs](https://github.com/StatCan/jupyter-apis/commit/0e116fd92e406d308f5474947bb8e9547127e11c)
    - [fix: set max cpu/ram values for 1 gpu + translation](https://github.com/StatCan/jupyter-apis/commit/6c5fa37258d6b3df7bafba5538655f64f5f3c5e9) 
  - [Merge pull request #57 from StatCan/lower-gpu-cpu-request](https://github.com/StatCan/jupyter-apis/commit/0b55d4da2ac2b43dbcbc0331a4d404366af6d788)
    - [fix: lower gpu node cpu-request to allow scheduling](https://github.com/StatCan/jupyter-apis/pull/57/commits/33372de4d74ec36b8e92c3b9edfa80ef4ab809ef) 

- [x] #81 -Rohan
  - [Merge pull request #9 from StatCan/fix-angular-component](https://github.com/StatCan/jupyter-apis/commit/f0b855dec4d375a0c06d521d404e4a6da4b8d30a)
    - [fix: angular mat-form-fiel error](https://github.com/StatCan/jupyter-apis/pull/9/commits/20ff2f912e7afc661116a5e4618fdd2bea016ad1)
  - [Merge pull request #23 from StatCan/fix-dependabot-alerts](https://github.com/StatCan/jupyter-apis/commit/1a50addb5e0703409dd5954d6aaeb68e35458b2c)
    - [fix: resolve dependabot alerts](https://github.com/StatCan/jupyter-apis/pull/23/commits/faf54151b9873bb42c8c7be8f5211e95c5bbb5e1)
    - [fix: browser console ExpressionChanged error](https://github.com/StatCan/jupyter-apis/pull/23/commits/d2a95378554f5523f626f947d0da37909f8f2548) 

- [x] https://github.com/StatCan/jupyter-apis/issues/69 Wendy + Saffa/Jose reviewing
  - [Merge pull request #30 from StatCan/Change-the-hint-image-message](https://github.com/StatCan/jupyter-apis/commit/9d02488af4a37971e76cca421ad6ab8ce96848c4)
    - [use Prettier](https://github.com/StatCan/jupyter-apis/pull/30/commits/52cc62b5efb230c00642890477c3272559845da5)
    - [Add message when shoose image](https://github.com/StatCan/jupyter-apis/pull/30/commits/a9e122a7776a92e8803a7f81c0140cd1d4872317)
  - [Merge pull request #60 from StatCan/image-selection-hint](https://github.com/StatCan/jupyter-apis/commit/5daefc35422e0d72cfa7c25326eaa1e4eeeebfc6)
    - [feat: url link to image info page](https://github.com/StatCan/jupyter-apis/commit/68c65461cd6002982cbc7e04bf7751fd0f50972d)

- [x] #83  Skye
  - [Merge pull request #61 from StatCan/59-add-custom-image-url-validation](https://github.com/StatCan/jupyter-apis/commit/416bcb15aa542232ebc0184b97f910d5d5faa96c)
    - [fix: added custom image url validation](https://github.com/StatCan/jupyter-apis/pull/61/commits/4e2554f77c0fb4528ae3edc4d8ddec876d098581)

- [x] https://github.com/StatCan/jupyter-apis/issues/75 Saffa + Rohan
  - [Merge pull request #27 from StatCan/kubecost-estimates](https://github.com/StatCan/jupyter-apis/commit/49dea695f64485252348b72608af0b7886d05cb6)
    - [feat: add cost estimate table](https://github.com/StatCan/jupyter-apis/pull/27/commits/b08cb8844caa2e57fba5ff29ff476a7177859c61)
    - [fix: connect to Kubecost API + cleanup and refactor components](https://github.com/StatCan/jupyter-apis/pull/27/commits/2284c4d49fd7c7dc13922ec07fd669b5bf1b2e8e)
  - [Merge pull request #34 from StatCan/fix-kubecost-error](https://github.com/StatCan/jupyter-apis/commit/d412a39fca66e4ae70c78742555c01f0e01ea11f)
    - [fix: add check for kubecost error response + translation](https://github.com/StatCan/jupyter-apis/pull/34/commits/460f113d6447e3541a8b9383c9b613386314b46f) 
  - [Merge pull request #40 from StatCan/fix-kubecost-message](https://github.com/StatCan/jupyter-apis/commit/0fba437b5f3965c141407dd5bd51ac2289570faa)
    - [fix: kubecost table message](https://github.com/StatCan/jupyter-apis/pull/40/commits/aeddf810fb0a20916ad4c2f990ad1a85ff0a54ee)

- [x] https://github.com/StatCan/jupyter-apis/issues/74 Skye
  - [Merge pull request #48 from StatCan/feat-env-var](https://github.com/StatCan/jupyter-apis/commit/2c1532b00791fce51d926d3a31a9f528179f7e72)
    - [feat: Add env variable to set language of the notebook server](https://github.com/StatCan/jupyter-apis/pull/48/commits/83af67391be036d279453a649bb5b6a9f08a985b)

- [x] https://github.com/StatCan/jupyter-apis/issues/71 Wendy
  - [Merge pull request #20 from StatCan/18-destroy-old-notebooks](https://github.com/StatCan/jupyter-apis/commit/2e9daf661793db54fddc1e22ad04746a90b4957d)
    - [fix: delete old notebooks (pvc)](https://github.com/StatCan/jupyter-apis/pull/20/commits/209518819360fcc760f46ac8ff5d89f14b844992)
    - [feat: Add icons to main tables + refactor](https://github.com/StatCan/jupyter-apis/pull/20/commits/fdb507f647288e6781cac7d68274acb5a7711bec)
    - [fix: Add status to pvc table + refactor](https://github.com/StatCan/jupyter-apis/pull/20/commits/49fc2b39df57eda0ffc32499e891cbd9b9195f37)
    - [feat: Convey more info via volume status column](https://github.com/StatCan/jupyter-apis/pull/20/commits/4fabd14e05dd91781a30fa24e76804fd77089b8d)
  - [Merge pull request #29 from StatCan/28-data-volume-ui-is-squashed](https://github.com/StatCan/jupyter-apis/commit/738e6fda2a4de40d6ac7114d4bf043b6069659dc)
    - [fix: Improve spacing for new data volume config](https://github.com/StatCan/jupyter-apis/pull/29/commits/8df1dc807d3f362523c3b23c1d44bd027faab542) 

- [x] https://github.com/StatCan/jupyter-apis/issues/72 Jose
   - [Merge pull request #51 from StatCan/50-add-validation-mounted-pvc](https://github.com/StatCan/jupyter-apis/commit/8641aabbd270e1c2d096a7c7e54f62050ae51048)
    - [feat: Add validation for mounted pvc](https://github.com/StatCan/jupyter-apis/pull/51/commits/f2614b68f6858c904b8b2f2292228e97efbb9b29)
    - [fix: show error when element is untouched](https://github.com/StatCan/jupyter-apis/pull/51/commits/22805a3f9dd5d39f3c99985d0bc68b9425f0fd53)

- [x] #84 (i.e. for the tag when a nb is prob, and also hvaing the option in the drop)

- [x] https://github.com/StatCan/jupyter-apis/issues/73 Wendy + someone else
  - [Merge pull request #35 from StatCan/32-angular-ngx-translate](https://github.com/StatCan/jupyter-apis/commit/f938db9495db669b280529482c82ff518effaec5)
    - [feat: Add Official Languages with ngx-translate](https://github.com/StatCan/jupyter-apis/pull/35/commits/2e213ecab0d43c97e5360dce6a508ba922d4399a)
    - will have to translate formatCost() to match cost format to French (in `index-default.component.ts`)

- [x] #78 -  https://github.com/StatCan/kubeflow/issues/51 Bring this back in since it wasn't included in 1.3 but we use it in prod - JOSE
- [x] #101 - After making the Kubecost changes, on dev there is an error in the cost response.

## Back end customizations:
- [x] support for start/stop notebooks
- [x] if we want, adjust our backend to support their new volumes page : We do not because of Pro-B
- [x] check for any new endpoints in 1.3 api that are not currently implemented in ours
    - [ ] jupyter (Skye)
    - [ ] common (Jose)
    - [ ] volumes

## TO TEST
- [x] Deleting anotebook ONLY deletes a notebook. 
- [x] Logos for the table
## TO DO:
- [x] New validation (including but not only) Limit CPU, limit Memory.
- [x] Fix up the i18n once all is merged

**Comment by wg102** [comment]
Is this something we want:
![image](https://user-images.githubusercontent.com/14837042/154754203-e79cfce4-f083-433b-aa57-77e2732e90a5.png)

**Comment by wg102** [comment]
Things to fix: 
![image](https://user-images.githubusercontent.com/14837042/156640649-9b0211e7-f112-41c9-acc8-7ffcf00c9eb8.png)
- [ ] No default language
- [ ] No i18n 
![image](https://user-images.githubusercontent.com/14837042/156640754-712c044e-d9b8-4513-a0e2-01a752d32948.png)
- [ ] Text not showing
![image](https://user-images.githubusercontent.com/14837042/156640841-b455bd15-7020-4e14-b4a9-6468d2808162.png)
- [ ] No negative validation
![image](https://user-images.githubusercontent.com/14837042/156640901-63e91fba-3954-4ddd-ad28-a89c661635ff.png)
- [ ] no i18n
![image](https://user-images.githubusercontent.com/14837042/156641006-b90d4398-3c74-47f2-9135-31dd855f9915.png)
- [ ] No validation for used volumes
![image](https://user-images.githubusercontent.com/14837042/156641091-fe7a7cfd-e3de-4287-a09f-56985e223fb7.png)
- no i18n

**Comment by wg102** [comment]
This was finished in April
